source/common/network/address_impl.h

```diff
@@ -358,6 +384,11 @@ class EnvoyInternalInstance : public InstanceBase {
   socklen_t sockAddrLen() const override { return 0; }
   absl::string_view addressType() const override { return "envoy_internal"; }
 
+  void abslHashValue(absl::HashState state) const override {
+    absl::HashState::combine(std::move(state), internal_address_.address_id_,
```

Comment:
Can you document why we know that equal EnvoyInternalInstance will have equal HashStates and unequal EnvoyInternalInstance will have unequal HashStates?

Reply:
Yea, I will fix that as above said.

Reply:
I will add doc.